### PR TITLE
Only add '\\usepackage[utf8]{inputenc}' when using pdflatex

### PR DESCRIPTION
**Pull request by avylove**
\usepackage[utf8]{inputenc} is not needed for lualatex and exlatex and breaks compiling when source files are UTF8 encoded.

This patch will only add the inputenc package when using pdflatex. In testing it seems it isn't needed, but it doesn't break compiling like it does with lualatex and exlatex.

For documentation change, specified change was in version 1.5 since it would be a change to current defaults.

Discusion of this topic can be found here:
http://tex.stackexchange.com/questions/44694/fontenc-vs-inputenc

### CODE REVIEW

**Comment by tk0miya** [comment]
Oh, I just merged this from accident before reviewing.

@jfbu Could you review this please?
It looks good to me. Actually, it works fine for all latex LaTeX engine; pdfTeX, LuaTeX, XeTeX and pLaTeX (for japanese) on my environment.

If it's okay, I will cherry-pick this into stable branch.

Note: It might be related with #2513.

**Comment by jfbu** [comment]
LGTM, I just point out that recent `\usepackage[utf8]{inputenc}` does _not_ break things under lualatex and xelatex as it then essentially does nothing:

```
Package inputenc Warning: inputenc package ignored with utf8 based engines.
```

Some years ago there were problems possibly however.

File: sphinx/writers/latex.py [hunk 1]
```diff
@@ -295,7 +296,9 @@ class LaTeXTranslator(nodes.NodeVisitor):
         'classoptions':    '',
         'extraclassoptions': '',
         'passoptionstopackages': '',
-        'inputenc':        '\\usepackage[utf8]{inputenc}',
+        'inputenc':        ('\\ifPDFTeX\n'
+                            '  \\usepackage[utf8]{inputenc}\n'
+                            '\\else\\fi'),
```
Comment (jfbu) [inline sphinx/writers/latex.py:301]: you could use `'\\fi'),` here, the `\\else` is not needed but I understand you use it before next lines do.

**Comment by tk0miya** [comment]
@jfbu Thank you for comment. I don't know old inputenc package crashes under lualatex and xelatex.
I'd overlooked that because I use latest TeXLive.

**Comment by tk0miya** [comment]
I cherry-picked this into stable. at same time, I removed `\\else` block.

Thank you all!

**Comment by jfbu** [comment]
@tk0miya

**Comment by jfbu** [comment]
(sorry for problems with my comments, I initially posted on wrong thread then I hit return key inadvertendly on this thread :-( , must urgently get strong coffee ;-) )

Brief look at the source of inputenc at

```
/usr/local/texlive/2016/texmf-dist/source/latex/base/inputenc.dtx
```

shows this

```
\changes{v1.2a}{2014/04/20}{Added alternative definition for xe/lua tex (pr/4024)}%
```

which was thus first introduced I guess into TeXLive 2014. And indeed, testing with TeXLive 2013 the following latex document:

``` latex
\documentclass{article}
\usepackage[utf8]{inputenc}
\begin{document}
éùà
\end{document}
```

compilation fails.

More recent versions of inputenc (as since TL2014) only print a warning.

**Comment by tk0miya** [comment]
Thank you for investigation.
This code will help some Linux distributions using a bit old TeXLive :-)

**Pull request merged**